### PR TITLE
Fixed yaw in NAV_LAND

### PR DESCRIPTION
**Pull request by tridge**
This fixes yaw handling in NAV_LAND in AUTO mode. The fixed yaw from a CONDITION_YAW was being ignored

### CODE REVIEW

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -876,7 +876,7 @@ void Copter::ModeAuto::land_run()
     // set motors to full range
     motors->set_desired_spool_state(AP_Motors::DESIRED_THROTTLE_UNLIMITED);
     
-    land_run_horizontal_control();
+    land_run_horizontal_control(auto_yaw.mode()==AUTO_YAW_FIXED, auto_yaw.yaw());
```
Comment (Aksellence) [inline ArduCopter/mode_auto.cpp:879]: Ah.. I see why we using this. But still my question lies, how were we able to compile the method without the params in the first place, as on looking at the repo I couldn't see any polymorphic function there (without the params)?

File: Tools/scripts/runcoptertest.py [hunk 2]
```diff
@@ -35,24 +35,25 @@ def wait_time(mav, simtime):
...
+mavproxy.send('param set SIM_SPEEDUP 5\n')
 wait_mode(mav, ['GUIDED'])
 mavproxy.expect('is using GPS')
 mavproxy.expect('is using GPS')
 mavproxy.send('arm throttle\n')
 mavproxy.expect('ARMED')
 mavproxy.send('mode AUTO\n')
 wait_mode(mav, ['AUTO'])
-wait_time(mav, 30)
+wait_time(mav, 15)
```
Comment (Aksellence) [inline Tools/scripts/runcoptertest.py:54]: Why did we reduce this time?

File: ArduCopter/mode_auto.cpp (hunk 1 above)
Comment (tridge) [reply ArduCopter/mode_auto.cpp:879]: it takes optional parameters (ie parameters with default values)

File: Tools/scripts/runcoptertest.py (hunk 2 above)
Comment (tridge) [reply Tools/scripts/runcoptertest.py:54]: this is just my little test script, 15s is closer to what we normally have between arming and takeoff. 30s was for when I was testing gps drift

Comment (Aksellence) [reply Tools/scripts/runcoptertest.py:54]: Okay Sounds good!

**Review by Aksellence — APPROVED** [review]

**Pull request merged**